optuna/storages/_rdb/storage.py

```diff
@@ -39,6 +41,43 @@
 _logger = optuna.logging.get_logger(__name__)
 
 
+class _ScopedSessionContextManager:
+    def __init__(self, scoped_session: orm.scoped_session) -> None:
+        self._scoped_session = scoped_session
+
+    def __enter__(self) -> orm.Session:
+        session = self._scoped_session()
+        return session
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[Exception]],
+        exc_value: Optional[Exception],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        try:
```

Comment:
If my understanding is correct, `__exit__` is called as soon as an exception is raised and propagated from within the context. In this case, we shouldn't try to commit but just roll back and then propagate the error further, i.e. return `None` here. How about adding a check in the beginning of this method and doing so?

Reply:
In the latest commit, I used `@contextmanager` instead of class.